zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -276,63 +344,140 @@ def channels_to_zerver_stream(slack_data_dir: str, realm_id: int, added_users: A
+                                                          subscription_id_count)
         zerver_recipient.append(recipient)
+        zerver_subscription.append(sub)
+        subscription_id_count += 1
+        recipient_id_count += 1
 
+    print('######### IMPORTING STREAMS FINISHED #########\n')
+    return zerver_defaultstream, zerver_stream, added_channels, zerver_subscription, \
+        zerver_recipient, added_recipient
+
+def build_defaultstream(channel_name: str, realm_id: int, stream_id: int) -> ZerverFieldsT:
+    if channel_name == "general":  # Slack specific
+        defaultstream = dict(
+            stream=stream_id,
+            realm=realm_id,
+            id=get_model_id(DefaultStream, 'zerver_defaultstream', 1))
+        return defaultstream
+    return None
+
+def build_pm_recipient_sub_from_user(zulip_user_id: int, recipient_id: int,
+                                     subscription_id: int) -> Tuple[ZerverFieldsT,
+                                                                    ZerverFieldsT]:
+    recipient = dict(
+        type_id=zulip_user_id,
+        id=recipient_id,
+        type=1)
+
+    sub = dict(
+        recipient=recipient_id,
+        notifications=False,
+        color="#c2c2c2",
+        desktop_notifications=True,
+        pin_to_top=False,
+        in_home_view=True,
+        active=True,
+        user_profile=zulip_user_id,
+        id=subscription_id)
+
+    return recipient, sub
+
+def build_subscription(channel_members: List[str], zerver_subscription: List[ZerverFieldsT],
+                       recipient_id: int, added_users: AddedUsersT,
+                       subscription_id: int) -> Tuple[List[ZerverFieldsT], int]:
+    for member in channel_members:
         sub = dict(
-            recipient=recipient_id_count,
+            recipient=recipient_id,
             notifications=False,
             color="#c2c2c2",
             desktop_notifications=True,
             pin_to_top=False,
             in_home_view=True,
             active=True,
-            user_profile=zulip_user_id,
-            id=subscription_id_count)
-
+            user_profile=added_users[member],
+            id=subscription_id)
+        # The recipient is a stream for stream-readable message.
+        # proof :  https://github.com/zulip/zulip/blob/master/zerver/views/messages.py#L240 &
+        # https://github.com/zulip/zulip/blob/master/zerver/views/messages.py#L324
         zerver_subscription.append(sub)
-        subscription_id_count += 1
-        recipient_id_count += 1
+        subscription_id += 1
+    return zerver_subscription, subscription_id
 
-    print('######### IMPORTING STREAMS FINISHED #########\n')
-    return zerver_defaultstream, zerver_stream, added_channels, zerver_subscription, \
-        zerver_recipient, added_recipient
+def slack_workspace_messages(slack_data_dir: str, REALM_ID: int, added_users: AddedUsersT,
```

Comment:
This function name should be in the form of a verb.

Reply:
@rht `get_slack_workspace_messages`?

Reply:
Yeah, while `convert_slack_workspace_messages_to_zulip_org_messages` would be too long.